block-node/publisher/src/main/java/org/hiero/block/node/publisher/BlockStreamProducerSession.java

```diff
@@ -174,9 +175,14 @@ void switchToBehind() {
         newBlockItems.clear();
         // let client know we do not need more data for the current block
         if (responsePipeline != null) {
-            final PublishStreamResponse behindResponse = new PublishStreamResponse(
-                    new OneOf<>(ResponseOneOfType.ACKNOWLEDGEMENT, new SkipBlock(currentBlockNumber)));
-            responsePipeline.onNext(behindResponse);
+            try {
+                final PublishStreamResponse behindResponse = new PublishStreamResponse(
+                        new OneOf<>(ResponseOneOfType.ACKNOWLEDGEMENT, new SkipBlock(currentBlockNumber)));
+                responsePipeline.onNext(behindResponse);
+            } catch (ClassCastException e) {
+                // temporary until we find out why it cannot cast skipblock to ack
+                LOGGER.log(WARNING, "Session {0} has thrown an exception: {1}", this, e.getMessage());
```

Comment:
Skip Block and Ack are not the same types, nor are they subtypes.
Skip Block is its own response type.
```suggestion
                final PublishStreamResponse behindResponse = new PublishStreamResponse(
                        new OneOf<>(ResponseOneOfType.SKIP_BLOCK, new SkipBlock(currentBlockNumber)));
                responsePipeline.onNext(behindResponse);
            } catch (ClassCastException e) {
                // temporary until we find out why it cannot cast skipblock to ack
                LOGGER.log(WARNING, "Session {0} has thrown an exception: {1}", this, e.getMessage());
```

Reply:
Yes, I fixed this, but stashed and forgot to push at the end of the day :D